Tests/NIOSSLTests/TLSConfigurationTest.swift

```diff
@@ -1077,6 +1096,50 @@ class TLSConfigurationTest: XCTestCase {
             XCTAssertEqual(Set([Wrapper(config: first), Wrapper(config: transformed)]).count, 2, "Should have hashed non-equal in index \(index)")
         }
     }
+    
+    func testObtainingTLSVersionOnClientChannel() throws {
+        
+        var clientConfig = TLSConfiguration.makeClientConfiguration()
+        clientConfig.maximumTLSVersion = .tlsv11
+        clientConfig.certificateVerification = .noHostnameVerification
+        clientConfig.trustRoots = .certificates([TLSConfigurationTest.cert1])
+
+        var serverConfig = TLSConfiguration.makeServerConfiguration(
+            certificateChain: [.certificate(TLSConfigurationTest.cert1)],
+            privateKey: .privateKey(TLSConfigurationTest.key1)
+        )
+        serverConfig.maximumTLSVersion = .tlsv11
+        serverConfig.certificateVerification = .none
+        
+        let clientContext = try assertNoThrowWithValue(NIOSSLContext(configuration: clientConfig))
+        let serverContext = try assertNoThrowWithValue(NIOSSLContext(configuration: serverConfig))
+
+        let group = MultiThreadedEventLoopGroup(numberOfThreads: 1)
```

Comment:
Rather than use MTELG can we use one of the `BackToBackEmbeddedChannel` constructions? These tests run much faster and are less flaky and weird.

Reply:
Sure, no problem. I added an initial refactor using `BackToBackEmbeddedChannel`, let me know your thoughts.